src/component/legend/LegendView.ts

```diff
@@ -404,7 +404,8 @@ class LegendView extends ComponentView {
         }
         else {
             // Use default legend icon policy for most series
-            const rotate = legendIconType === 'inherit' && seriesModel.getData().getVisual('symbol')
+            const rotate = (!legendIconType || legendIconType === 'inherit')
```

Comment:
I think this may not be quite accurate.

Consider the series that doesn't have `seriesModel.getLegendIcon`, like pie series, with `legend.symbolRotate = 30`, we may probably expect the legend icon is a `roundRect` with rotation of 30 degrees. But instead, the current implementation uses 0. I didn't consider this situation when in [my previous PR](https://github.com/apache/echarts/commit/b65e590a46c884c480c1eab2fc946927a203a421#diff-57be669601679875d9b233c7dc218568caae93a25cf1585ee5df4129dcfd82a2R384). Can you help fix this?

My guess would be something like `zrUtil.retrive2(seriesModel.getData().getVisual('symbolRotate'), iconRotate)` but you may need to figure out what works by doing more tests.

Thanks!

Reply:
OK I will test more.